### DISCUSSION TITLE
Configure routing with lazy-loading of tiles over HTTP

### DISCUSSION THREAD

**Discussion by harto** [valhalla/valhalla#4703] [Q&A] [answered]
Hello,

https://github.com/valhalla/valhalla/discussions/3323#discussioncomment-1330823 mentions that Valhalla can pull tiles on demand over HTTP, as an alternative to providing a tarball at startup.

I have a tileset that I built for Australia that works as expected when packaged as a tarball and dropped into `/data/valhalla/tiles.tar`. But when I untar that same file and serve it over HTTP, and configure Valhalla to fetch from the corresponding URL, I get the following error:
```
Locations are in unconnected regions. Go check/edit the map at osm.org
```

I can see that the tiles are being downloaded to `/data/valhalla/…`. But it doesn't seem like they can be used correctly.

The working deployment is started with a completely vanilla config, i.e.:
```
valhalla_build_config >config.json
valhalla_service config.json
```

The non-working version is started like this:
```
valhalla_build_config --mjolnir-tile-url "http://example.com/australia/{tilePath}" >config.json
valhalla_service config.json
```

Did I miss something?

In case it's relevant:
- The tiles are being hosted in an S3 bucket.
- The Valhalla service is running on AWS ECS.

Thanks :-)

**Comment by nilsnolde** [comment, marked as answer]
I'm pretty sure this has to do with the connectivity map, which can't be correct because you don't have a resident graph. try with `loki.use_connectivity: false` in the json config. possibly there's a few more gotchas like this. I personally never tried it myself.

**Reply by harto** [reply]
OK yes, thank you! That does seem to fix it.

What are the implications of disabling the connectivity map? Does that impact performance? My hope is that aside from the initial cost of downloading tiles, performance will be equivalent to having the full tileset tarball available locally. But it would be helpful to know if that is not realistic.

**Reply by nilsnolde** [reply]
Once they're downloaded it's fine. Though there's other drawbacks which might be non-obvious, such as the tar is mmapped while tile dir isn't. meaning each thread/process (depending how you run valhalla) will need their own full-blown tile cache, while in the tar case mmapping is providing that mostly.

Disabling the connectivity map does hurt a bit for cases where stuff obviously isn't connected, like going from north to south america (can't cross the darien gap). then the algorithm just expands until it fails instead of failing early because it knows there's no connection..

**Reply by harto** [reply]
Great. Thanks again!
